blackbird_python/blackbird/program.py

```diff
@@ -363,8 +384,12 @@ def serialize(self):
                         array_insert += len(bb_array)
 
                     elif isinstance(v, str):
-                        # argument is a string type
-                        args.append('"{}"'.format(v))
+                        # argument is a string type; if a p-type parameter (e.g. p0),
+                        # then simply add it as is
+                        if v[0] == "p":
+                            args.append(v)
```

Comment:
Perhaps this should also be, e.g.,
```suggestion
                        if v[0] == "p" and self.programtype["name"] == "tdm":
                            args.append(v)
```